internal/services/storage/storage_account_resource.go

```diff
@@ -2275,6 +2404,22 @@ func resourceStorageAccountDelete(d *pluginsdk.ResourceData, meta interface{}) e
 	return nil
 }
 
+func handleStorageServiceError(operation string, service string, err error) (interface{}, string, error) {
+	// if the error is a DetailedError type, check the status code and return the appropriate state
+	var respErr azautorest.DetailedError
+	if errors.As(err, &respErr) {
+		log.Printf("[DEBUG] error %s %s service properties: method=%s, statusCode=%d, message=%s\n", operation, service, respErr.Method, respErr.StatusCode.(int), respErr.Message)
+		if respErr.StatusCode.(int) == 404 {
+			// if the status code is 404 (not found), retry the request
+			log.Printf("[DEBUG] %s service properties not available, retrying...\n", service)
+			return nil, "Pending", nil
+		}
+	}
+	// if the error is unhandled type or status, log the error and return the error state
+	log.Printf("[DEBUG] unexpected error while %s %s service properties: %v\n", operation, service, err)
+	return nil, "Error", err
+}
```

Comment:
we're currently moving off of `Azure/azure-sdk-for-go` and `Azure/go-autorest` - as such casting to those error types isn't something we can rely on long-term - however we should be able to check the status code here via:

```suggestion
func handleStorageServiceError(operation string, service string, err error) (interface{}, string, error) {
	// if the error is a DetailedError type, check the status code and return the appropriate state
	var respErr azautorest.DetailedError
	if errors.As(err, &respErr) {
		log.Printf("[DEBUG] error %s %s service properties: method=%s, statusCode=%d, message=%s\n", operation, service, respErr.Method, respErr.StatusCode.(int), respErr.Message)
		if response.WasNotFound(respErr.StatusCode.(int)) {
			// if the status code is 404 (not found), retry the request
			log.Printf("[DEBUG] %s service properties not available, retrying...\n", service)
			return nil, "Pending", nil
		}
	}
	// if the error is unhandled type or status, log the error and return the error state
	log.Printf("[DEBUG] unexpected error while %s %s service properties: %v\n", operation, service, err)
	return nil, "Error", err
}
```

Reply:
Here, the `if response.WasNotFound()` expect `*http.response` not the status

what i can do is `if response.WasNotFound(respErr.Response)` however the `Azure/go-autorest` will be still in place

I also checked the client set/update methods. All return response in response properties except for queue service. Queue service return `autorest.DetailedError` only, so this one will require the `Azure/go-autorest` anyway.

I would suggest to migrate it once `Azure/azure-sdk-for-go` will be gone.